### PR TITLE
Migrate URIBuilder to httpcore5, Fixes AB#2909378

### PR DESCRIPTION
**Pull request by rpdome**
Part 1 of the work to get rid msebera.httpclient. 
[AB#2909378](https://identitydivision.visualstudio.com/fac9d424-53d2-45c0-91b5-ef6ba7a6bf26/_workitems/edit/2909378)

Migrate the usage of uribuilder to httpcore5

### CODE REVIEW

**Review by fadidurah — APPROVED** [review]

File: common4j/build.gradle [hunk 1]
```diff
@@ -219,6 +219,7 @@ dependencies {
     implementation "org.json:json:$rootProject.ext.jsonVersion"
     implementation "com.github.stephenc.jcip:jcip-annotations:$rootProject.ext.jcipAnnotationVersion"
     implementation "cz.msebera.android:httpclient:$rootProject.ext.mseberaApacheHttpClientVersion"
```
Comment (p3dr0rv) [inline common4j/build.gradle:221]: so are we removing msebera in a separate PR?

File: common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java [hunk 2]
```diff
@@ -99,9 +99,8 @@ public static URL appendPathAndQueryToURL(@NonNull final URL urlToAppend,
         }
 
         builder.setPathSegments(pathSegments);
-        if (queryParam != null && !queryParam.isEmpty()) {
-            builder.setQuery(queryParam);
-        }
+        builder.addParametersIfAbsent(queryParam);
```
Comment (p3dr0rv) [inline common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java:102]: now that we remove the check do, we have tests for null or empty map?

**Review by p3dr0rv — APPROVED** [review]
LGTM, leave a few questions

File: common4j/build.gradle (hunk 1 above)
Comment (rpdome) [reply common4j/build.gradle:221]: yep. There's part 2.

File: common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java (hunk 2 above)
Comment (rpdome) [reply common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java:102]: addParametersIfAbsent already handles null or empty map.

Comment (rpdome) [reply common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java:102]: added tests in CommonURIBuilderTests.

**Pull request merged**